include/fastdds/rtps/common/SerializedPayload.h

```diff
@@ -124,6 +124,10 @@ struct RTPS_DllAPI SerializedPayload_t
             }
         }
         encapsulation = serData->encapsulation;
+        if(!data || !serData->data)
+        {
+            return false;
+        }
```

Comment:
Those pointers can only be `nullptr` if the length of the payload is 0. Otherwise, a bad_alloc is thrown. When the length is 0, it should return `true`, as the copy has been successful.

```suggestion
        if (length == 0)
        {
            return true;
        }
```